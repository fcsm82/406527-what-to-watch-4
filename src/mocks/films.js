import {getRandomElement, generateUniqueId} from "../utils/utils.js";

const FILMS_NUMBER = 8;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];

const POSTERS = [
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `bohemian-rhapsody.jpg`,
  `macbeth.jpg`,
  `aviator.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
  `revenant.jpg`,
  `johnny-english.jpg`,
  `shutter-island.jpg`,
  `pulp-fiction.jpg`,
  `no-country-for-old-men.jpg`,
  `snatch.jpg`,
  `moonrise-kingdom.jpg`,
  `seven-years-in-tibet.jpg`,
  `midnight-special.jpg`,
  `war-of-the-worlds.jpg`,
  `dardjeeling-limited.jpg`,
  `orlando.jpg`,
  `mindhunter.jpg`,
  `midnight-special.jpg`
];

export const TOP_FILM = {
  title: `The Grand Budapest Hotel`,
  poster: `the-grand-budapest-hotel-poster.jpg`,
  genre: `Drama`,
  releaseDate: 2014
};

export const createFilms = () => {
  let films = [];
  for (let i = 0; i < FILMS_NUMBER; i++) {
    films.push({
      id: generateUniqueId(),
      title: getRandomElement(TITLES),
      poster: getRandomElement(POSTERS),
    });
  }
  return films;
};
