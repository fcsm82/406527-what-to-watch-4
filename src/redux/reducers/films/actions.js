export const Action = {
  CHANGE_FILTER: `Change filter`,
  SHOW_MORE: `Show more films`,
};

export const Creator = {
  filterFilms: (filter) => {
    return {
      type: Action.CHANGE_FILTER,
      payload: filter
    };
  },

  showMore: () => {
    return {
      type: Action.SHOW_MORE,
    };
  }
};
