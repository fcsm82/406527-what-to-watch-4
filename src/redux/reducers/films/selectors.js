// import {createSelector} from "reselect";
import NameSpace from "../name-space.js";
// import {getUniqueGenres} from "../../../utils/utils.js";
// import {getFilms} from "../data/selectors.js";

// export const getFilters = createSelector(
//     getFilms,
//     (films) => getUniqueGenres(films)
// );

export const getActiveFilter = (state) => {
  return state[NameSpace.FILMS].activeFilter;
};

export const getShownFilms = (state) => {
  return state[NameSpace.FILMS].shownFilms;
};

export const getView = (state) => {
  return state[NameSpace.FILMS].view;
};
