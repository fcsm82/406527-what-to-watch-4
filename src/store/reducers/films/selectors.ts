
import NameSpace from "../name-space";

export const getActiveFilter = (state) => {
  return state[NameSpace.FILMS].activeFilter;
};

export const getShownFilms = (state) => {
  return state[NameSpace.FILMS].shownFilms;
};
