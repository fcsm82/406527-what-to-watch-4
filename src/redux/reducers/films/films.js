
import {Action} from "../../actions.js";
// import {View} from "../../../components/app/app.jsx";
import {FILTER} from "../../../consts.js";

// import {getPromoFilm, getFilms} from "../data/selectors.js";

const initialState = {
  // view: View.LIST,
  shownFilms: 8,
  activeFilter: FILTER.ALL,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: 8
      });

      // case Action.CHANGE_VIEW:
      //   return Object.assign({}, state, {
      //     view: View.DETAILS,
      //     activeFilm: action.payload
      //   });

      // case Action.CHANGE_VIEW_TO_SIGN_IN:
      //   return Object.assign({}, state, {
      //     view: View.SIGN_IN,
      //   });

    case Action.RENDER_FILMS:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + 8
      });

      // case Action.PLAY_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.VIDEO
      //   });

      // case Action.EXIT_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.LIST
      //   });

    default:
      return state;
  }
};

export {reducer};
