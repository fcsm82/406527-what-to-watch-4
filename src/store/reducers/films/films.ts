
import {Action} from "./actions";
import {ALL_GENRES} from "../../../consts";

const FilmsCount = {
  ON_START: 8,
  SHOW_MORE: 8
};

const initialState = {
  shownFilms: FilmsCount.ON_START,
  activeFilter: ALL_GENRES,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: FilmsCount.ON_START
      });

    case Action.SHOW_MORE:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + FilmsCount.SHOW_MORE
      });

    default:
      return state;
  }
};

export {reducer};
