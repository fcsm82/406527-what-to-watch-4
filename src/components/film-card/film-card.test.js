import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film-card.jsx";

import {film} from "../../mocks/films.js";

it(`Render FilmCard`, () => {
  const tree = renderer
    .create(<FilmCard
      film = {film}
      onTitleClick = {() => {}}
      onPosterClick = {() => {}}
      isPlaying = {true}
      setPlayingFilm = {() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
