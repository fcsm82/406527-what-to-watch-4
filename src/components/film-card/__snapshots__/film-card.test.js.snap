// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FilmCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        muted={true}
        poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Gangs of new york
    </a>
  </h3>
</article>
`;
